drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -167,6 +168,22 @@ def mci_id
     ac_hmis_mci_ids.to_a.min_by(&:id)&.value
   end
 
+  def update_mci_client
+    return unless ac_hmis_mci_ids.exists?
```

Comment:
If you implement my suggestion above, we can remove this. We don't want to run this query every time the client is saved. The job will throw an error if the client doesnt have an MCI ID. (which shouldn't happen if we only invoke this actio based on a `update_mci_attributes` attributes.